data/card_definitions.json

```diff
@@ -7409,6 +7409,340 @@
 		"NOTEEND": "MAY CARDS END"
 	},
 
+	{
+		"NOTE": "MIKA CARDS START",
+		"id": "mika_revolution",
+		"type": "ultra",
+		"display_name": "Mika's Revolution",
+		"force_cost": 0,
+		"gauge_cost": 2,
+		"range_min": 1,
+		"range_max": 2,
+		"speed": 2,
+		"power": 3,
+		"armor": 3,
+		"guard": 2,
+		"effects": [
+			{
+				"timing": "hit",
+				"effect_type": "push",
+				"amount": 7,
+				"and": {
+					"effect_type": "close",
+					"amount": 7,
+					"save_spaces_as_strike_x": true,
+					"and": {
+						"effect_type": "powerup",
+						"amount": "strike_x"
+					}
+				}
+			}
+		],
+		"boost": {
+			"boost_type": "continuous",
+			"force_cost": 1,
+			"cancel_cost": -1,
+			"display_name": "Diesel",
+			"effects": [
+				{
+					"timing": "during_strike",
+					"effect_type": "powerup",
+					"amount": 4
+				},
+				{
+					"timing": "start_of_next_turn",
+					"effect_type": "draw",
+					"amount": 2,
+					"and": {
+						"effect_type": "add_to_gauge_immediately"
+					}
+				}
+			]
+		}
+	},
+	{
+		"id": "mika_galaxy",
+		"type": "ultra",
+		"display_name": "Mika's Galaxy",
+		"force_cost": 0,
+		"gauge_cost": 3,
+		"range_min": 0,
+		"range_max": 1,
+		"speed": 1,
+		"power": 12,
+		"armor": 0,
+		"guard": 4,
+		"effects": [
+			{
+				"timing": "during_strike",
+				"effect_type": "calculate_range_from_center"
+			},
+			{
+				"timing": "before",
+				"condition": "was_wild_swing",
+				"effect_type": "choice",
+				"choice": [
+					{ "effect_type": "push", "amount": 2 },
+					{ "effect_type": "pass" }
+				]
+
+			}
+		],
+		"boost": {
+			"boost_type": "immediate",
+			"force_cost": 0,
+			"cancel_cost": -1,
+			"display_name": "Smart Steer",
+			"effects": [
+				{
+					"timing": "immediate",
+					"effect_type": "choice",
+					"choice": [
+						{ "effect_type": "advance", "amount": 1 },
```

Comment:
maybe it is time to change these into like "move_to_any_space" but with a "move_min": 0, "move_max": 3 or something.
That'll bring up the space selection UI instead of the choice box and probably be a better overall experience.

Reply:
Figuring out exactly which spaces to allow will be a little annoying since you have to consider retreat into the edge and account for the opponent, but I think it'll be better in the end.